Jenkinsfile

```diff
@@ -241,50 +241,33 @@ void postFailureFlexcodeScanAndReport(def vulnerabilities){
-        sendMail email, "</h3> $body", false, " high vulnerabilities in stage ${STAGE_NAME} ",'inventoriesRprt.report.json'
-    }
-    else {
-        sendMail email,"<h3>Pipeline FAILED in stage ${STAGE_NAME} <a href=${BUILD_URL}/console> Check Console Output Here</a></h3> ",false," pipeline FAILED in ${STAGE_NAME} "
-    }
-}
-
-def flexcodeScanAndReport(){
-
     def palamida_url="http://palamida-2.marklogic.com:8888"
-    def inventoriesRprt=''
-
     cleanWs deleteDirs: true, patterns: [[pattern: 'data-hub/**', type: 'EXCLUDE']]
 
     withCredentials([string(credentialsId: 'palamida_jwt', variable: 'jwt')]) {
 
         StartScan(baseUrl: palamida_url, projectName: 'ml-dhf', token: jwt)
 
-        inventoriesRprt = sh(returnStdout: true, script: '''curl -X GET ''' + palamida_url + '''/codeinsight/api/project/inventory/21 -H "Authorization: Bearer $jwt" ''')
-    }
+        sh '''
 
-    writeFile(file: 'inventoriesRprt.report.json', text: inventoriesRprt)
+         report_json="marklogic-data-hub-report.json"
+         report_p1="marklogic-data-hub-report-p1.json"
+         email_body="email-report.json"
 
-    def slurper = new JsonSlurperClassic().parseText(inventoriesRprt.toString().trim())
+         curl -X GET ''' + palamida_url + '''/codeinsight/api/project/inventory/21 -H "Authorization: Bearer $jwt" > $report_json
 
-    def vulnerabilities = []
-    slurper.inventoryItems.each { key, value ->
-        key.vulnerabilities.each { key1, value1 ->
-            if(key1.vulnerabilityCvssV3Severity == 'CRITICAL' || key1.vulnerabilityCvssV2Severity == 'HIGH'){
-                def itnumb = ((key.itemNumber as Integer) - 1)
-                def msg = " item:$itnumb; inventory name:$key.name; volnarability name:$key1.vulnerabilityName; with Severity V2:$key1.vulnerabilityCvssV2Severity; with Severity V3:$key1.vulnerabilityCvssV3Severity; "
-                vulnerabilities.add(msg)
-            }
-        }}
+         jq '[[.inventoryItems | .[] | select(.vulnerabilities != null) | select(.vulnerabilities[].vulnerabilityCvssV3Severity == "CRITICAL" or .vulnerabilities[].vulnerabilityCvssV2Severity == "HIGH")] | unique | .[] | del(.vulnerabilities[] | select(.vulnerabilityCvssV3Severity != "CRITICAL" and .vulnerabilityCvssV2Severity != "HIGH"))]' $report_json>$report_p1
 
-    archiveArtifacts artifacts: 'inventoriesRprt.report.json'
+         if [ "$(cat $report_p1)" != '[]' ]
+         then
+          cat $report_p1 | jq '.[] | {(.name): (.vulnerabilities+.filePaths)}' | sed '1d;$d;/vulnerabilityId/d;/vulnerabilityDescription/d;/vulnerabilityCvssV.Score/d;/vulnerabilitySource/d;/^{/d;/^}/d' | sed '1s/^/******** HIGH VULNERABILITIES: \\n\\n/;'>$email_body
+          mailx -s "FlexCode high vulnerabilities" ''' + email + ''' < $email_body
```

Comment:
Can we use jenkins email plugin instead of mailx? I'm not sure if mailx is installed on all the machines or not.

Reply:
i fixed it with original Jenkins plugin `emailext` function. 
Can't use shared library `sendMail` because `sendMail` doesn't have control over `mimeType` parameter.